### DISCUSSION TITLE
Allow users to set colormap of visuals consistent with matplotlib

### DISCUSSION THREAD

**Discussion by loewenm** [optuna/optuna#5357] [Ideas]
Please include the ability for users to set the colormap of visuals in a consistent way with matplotlib.

Reference: https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.set_cmap.html

The blue and white cmap theme used is _nice_ but quite limiting for broader appeal.

**Comment by nzw0301** [comment]
[Since Matplotlib based visualization module is an editable object](https://optuna.readthedocs.io/en/stable/tutorial/10_key_features/005_visualization.html#customize-generated-figures), we can change the coluormap manually.

**Comment by loewenm** [comment]
@nzw0301 Thanks for the reply. I realize that update_layout returns a plotly editable object.

However, I'm having difficulty with editing the colormap via update_layout(). 

For example: 
```
color_scale = dict([[0, 'green'], [0.5, 'red'], [1.0, 'rgb(0, 0, 255)']])
fig = optuna.visualization.plot_parallel_coordinate(loaded_study)
fig.update_layout({"width": 1600, "height": 400}, colorscale = color_scale)
```

Returns the following error:
```
TypeError: object of type 'int' has no len()
```

I've defined a dictionary of the desired color scale as per the documentation...

Consequently, I'm requesting that we are able to edit colors via a cmap consistent with matplotlib.

**Reply by nzw0301** [reply]
> I've defined a dictionary of the desired color scale as per the documentation...

I think this issue is unrelated to optuna, so the ploty community is more appropriate place to ask the question. 

For matplotlib, could you share the code and clarify which colour you'd like to change?
